core/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineRequest.java

```diff
@@ -135,15 +136,34 @@ static Parsed parseWithPipelineId(String pipelineId, Map<String, Object> config,
         if (pipeline == null) {
             throw new IllegalArgumentException("pipeline [" + pipelineId + "] does not exist");
         }
-        List<IngestDocument> ingestDocumentList = parseDocs(config);
-        return new Parsed(pipeline, ingestDocumentList, verbose);
+        boolean success = false;
+        try {
+            List<IngestDocument> ingestDocumentList = parseDocs(config);
+            Parsed parsed = new Parsed(pipeline, ingestDocumentList, verbose);
+            success = true;
+            return parsed;
+        } finally {
+            if (success == false) {
+                IOUtils.closeWhileHandlingException(pipeline);
+            }
+        }
     }
 
     static Parsed parse(Map<String, Object> config, boolean verbose, PipelineStore pipelineStore) throws Exception {
         Map<String, Object> pipelineConfig = ConfigurationUtils.readMap(null, null, config, Fields.PIPELINE);
-        Pipeline pipeline = PIPELINE_FACTORY.create(SIMULATED_PIPELINE_ID, pipelineConfig, pipelineStore.getProcessorFactories());
-        List<IngestDocument> ingestDocumentList = parseDocs(config);
-        return new Parsed(pipeline, ingestDocumentList, verbose);
+        Pipeline pipeline = null;
+        boolean success = false;
+        try {
+            pipeline = PIPELINE_FACTORY.create(SIMULATED_PIPELINE_ID, pipelineConfig, pipelineStore.getProcessorFactories());
```

Comment:
This can be outside the try/catch right? If there is a failure to create the pipeline, there is no pipeline to close.